### ISSUE TITLE
[UWP][Bug] SearchBar placeholder, and potentially text, overlaps SearchIcon

### ISSUE THREAD

**Issue by adammeaney** [xamarin/Xamarin.Forms#8784]
### Description
As shown in the following screenshot, when the placeholder is large enough to cover the entire SearchBar, it overlaps with the Search Icon.

### Steps to Reproduce

1.  Have a searchbar
2.  Give it a long placeholder
3.  Resize the window until there is overlap, if necessary.

### Expected Behavior
Placeholder elipsizes when intersecting search icon.

### Actual Behavior
Overlaps.

### Basic Information

- Version with issue: Forms 4.2 and above at least
- Last known good version: Unknown
- IDE: VS 2019 16.3.10
- Platform Target Frameworks: 
  - UWP:  16299

### Screenshots

![image](https://user-images.githubusercontent.com/6599983/70332331-f8b98680-1806-11ea-93c2-b468aa70f5ac.png)

### Reproduction
```XAML
 <Grid>
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="200"></ColumnDefinition>
            <ColumnDefinition Width="*"></ColumnDefinition>
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
            <RowDefinition Height="Auto"></RowDefinition>
            <RowDefinition Height="*"></RowDefinition>
        </Grid.RowDefinitions>
        <SearchBar WidthRequest="100" Placeholder="awefhawefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh awefh  "></SearchBar>        
        
    </Grid>
```

**Comment by WayaFlyfeather** [comment]
Note that this is actually not an issue in 4.3 at the moment, because of #8743 - but it will be when fixed (perhaps with the merge of #8773).

**Comment by WayaFlyfeather** [comment]
I've got a fix for this, but I've made it off of #8773 - so will make a PR once that is approved/merged.

The overlap is only an issue for the Placeholder, not Text (which uses a ScrollViewer).
